internal/transports/p2p/peer/peer.go

```diff
@@ -75,6 +105,39 @@ func NewPeer(
 
 	peer := &Peer{
 		addr:              netAddr,
+		inbound:           false,
+		cfg:               cfg,
+		chainParams:       chainParams,
+		headersService:    headersService,
+		chainService:      chainService,
+		checkpoints:       chainParams.Checkpoints,
+		nextCheckpoint:    nextCheckpoint,
+		log:               log,
+		services:          wire.SFspv,
+		protocolVersion:   initialProtocolVersion,
+		syncedCheckpoints: nextCheckpoint == nil,
+		wg:                sync.WaitGroup{},
+		msgChan:           make(chan wire.Message, writeMsgChannelBufferSize),
+		quitting:          false,
+		quit:              make(chan struct{}),
+	}
+	return peer, nil
+}
+
+func NewInboundPeer(
+	listener net.Listener,
+	cfg *config.P2PConfig,
+	chainParams *chaincfg.Params,
```

Comment:
This is idea for another task, but maybe we should make chainParams a part of P2PConfig instead of retrieving it the way we currently do, WDYT?

Reply:
You're right, I've created a task for that